src/helm/proxy/clients/megatron_client.py

```diff
@@ -0,0 +1,95 @@
+import json
+import requests
+from typing import Any, Dict, List
+import traceback
+
+from helm.common.cache import CacheConfig
+from helm.common.request import EMBEDDING_UNAVAILABLE_REQUEST_RESULT, Request, RequestResult, Sequence, Token
+from helm.common.tokenization_request import TokenizationRequest
+from helm.proxy.clients.huggingface_client import HuggingFaceClient
+from helm.proxy.clients.client import Client, wrap_request_time, truncate_sequence
+
+
+class MegatronClient(HuggingFaceClient):
+    def __init__(self, cache_config: CacheConfig):
+        super().__init__(cache_config)
+
+    def _send_request(self, raw_request: Dict[str, Any]) -> Dict[str, Any]:
+        response = requests.request(
+            method="PUT",
+            # TODO(tgale): Make this configurable.
+            url="http://localhost:5000/api",
+            headers={
+                "Content-Type": "application/json; charset=UTF-8",
+            },
+            data=json.dumps(raw_request),
+        )
+        out = json.loads(response.text)
+
+        # Detect if the server returned an error string.
+        if type(out) != dict:
+            raise ValueError(f"{response}: {response.text}")
+        return out
+
+    def _tokenize_response(self, text: str) -> List[Token]:
+        tokenized_text = self.tokenize(TokenizationRequest(text, tokenizer="huggingface/gpt2"))
+
+        # TODO(tgale): Support logprobs.
+        tokens = [Token(text=str(token), logprob=0, top_logprobs={}) for token in tokenized_text.raw_tokens]
+        return tokens
+
+    def _make_request(self, request: Request) -> RequestResult:
+        # Embedding not supported for this model
+        if request.embedding:
+            return EMBEDDING_UNAVAILABLE_REQUEST_RESULT
+
+        # TODO(tgale): Relax these.
+        assert request.num_completions == 1
+        assert not request.echo_prompt
+        assert not request.stop_sequences
+        assert request.top_p == 1
+
+        # TODO(tgale): Handle log probabilities.
+        raw_request = {
+            "prompts": [request.prompt],
+            "tokens_to_generate": request.max_tokens,
+            "temperature": 1e-7 if request.temperature == 0 else request.temperature,
```

Comment:
You should just set `"temperature": request.temperature`.

Add `BUGGY_TEMP_0_TAG` to your model tags and HELM will automatically add a small quantity to zero temperatures for you. See #1473

Reply:
Done.